platform-sdk/swirlds-logging-log4j-appender/src/main/java/com/swirlds/logging/log4j/appender/SwirldsLogAppender.java

```diff
@@ -0,0 +1,198 @@
+ *               </Root>
+ *             </Loggers>
+ *         </Configuration>
+ *        }
+ *     </pre>
+ */
+@Plugin(name = SwirldsLogAppender.APPENDER_NAME, category = "Core", elementType = "appender", printObject = true)
+public class SwirldsLogAppender extends AbstractAppender implements LogProvider {
+    /**
+     * The name of the appender for Log4j2 configuration.
+     */
+    public static final String APPENDER_NAME = "SwirldsLoggingAppender";
+
+    /**
+     * The log event factory to create log events.
+     * This is set by the swirlds-logging API.
+     */
+    private static LogEventFactory logEventFactory;
+    /**
+     * The log event consumer to consume log events.
+     * This is set by the swirlds-logging API.
+     */
+    private static LogEventConsumer logEventConsumer;
+
+    /**
+     * Constructs a new SwirldsLogAppender instance with default values.
+     * This constructor is used by the swirlds-logging API.
+     */
+    public SwirldsLogAppender() {
+        super(APPENDER_NAME, null, null, false, Property.EMPTY_ARRAY);
+    }
+
+    /**
+     * Constructs a new SwirldsLogAppender instance.
+     * This constructor is used by the Log4j2 framework.
+     *
+     * @param name   The name of the appender.
+     * @param filter The filter to apply.
+     * @param layout The layout of log messages.
+     */
+    protected SwirldsLogAppender(final String name, final Filter filter, final Layout<? extends Serializable> layout) {
+        super(name, filter, layout, false, Property.EMPTY_ARRAY);
+    }
+
+    /**
+     * Factory method to create a SwirldsLogAppender instance.
+     * This method is used by the Log4j2 framework.
+     *
+     * @param name   The name of the appender.
+     * @param layout The layout of log messages.
+     * @param filter The filter to apply.
+     *
+     * @return A new instance of SwirldsLogAppender.
+     */
+    @PluginFactory
+    public static SwirldsLogAppender createAppender(
+            @PluginAttribute("name") final String name,
+            @PluginElement("Layout") final Layout<? extends Serializable> layout,
+            @PluginElement("Filters") final Filter filter) {
+        return new SwirldsLogAppender(name, filter, layout);
+    }
+
+    /**
+     * If the log provider was installed from the swirlds-logging API,
+     * {@code event} will be forwarded to the swirlds-logging API.
+     *
+     * @param event The log event to append.
+     * @see SwirldsLogAppender#install(LogEventFactory, LogEventConsumer)
+     */
+    @Override
+    public void append(final LogEvent event) {
+        if (logEventFactory != null && logEventConsumer != null) {
+            logEventConsumer.accept(logEventFactory.createLogEvent(
+                    translateLevel(event.getLevel()),
+                    event.getLoggerName(),
+                    event.getThreadName(),
+                    event.getTimeMillis(),
+                    event.getMessage().getFormattedMessage(),
```

Comment:
that is inefficient since it will happen even if the message is never printed. You should provide a custom message implementation of our interface